frontend/web/components/pages/GitHubSetupPage.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { FC, useState, useEffect } from 'react'
+import OrganisationSelect from 'components/OrganisationSelect'
+import Input from 'components/base/forms/Input'
+import Icon from 'components/Icon'
+import InputGroup from 'components/base/forms/InputGroup'
+import ProjectFilter from 'components/ProjectFilter'
+import Utils from 'common/utils/utils'
+import Button from 'components/base/forms/Button'
+import { useCreateGithubIntegrationMutation } from 'common/services/useGithubIntegration'
+import { useCreateGithubRepositoryMutation } from 'common/services/useGithubRepository'
+import { useGetGithubReposQuery } from 'common/services/useGithub'
+import PanelSearch from 'components/PanelSearch'
+
+type Location = {
+  search: string
+}
+
+type GitHubSetupPageType = {
+  location: Location
+}
+
+type projectType = {
+  value: string
+  label: string
+}
+
+type repoType = {
+  value: string
+  label: string
+  name?: string
+}
+
+const GitHubSetupPage: FC<GitHubSetupPageType> = ({ location }) => {
+  const installationId =
+    new URLSearchParams(location.search).get('installation_id') || ''
+  const [organisation, setOrganisation] = useState<string>('')
+  const [project, setProject] = useState<any>({})
+  const [projects, setProjects] = useState<projectType[]>([])
+  const [repositoryName, setRepositoryName] = useState<string>('')
+  const [repositoryOwner, setRepositoryOwner] = useState<string>('')
+  const [repositories, setRepositories] = useState<any>([])
+
+  const { data: repos, isSuccess: reposLoaded } = useGetGithubReposQuery(
+    {
+      installation_id: installationId,
+    },
+    { skip: !installationId },
+  )
+
+  const [
+    createGithubIntegration,
+    { data, isSuccess: isSuccessCreateGithubIntegration },
+  ] = useCreateGithubIntegrationMutation()
+  const baseUrl = window.location.origin
+
+  const [
+    createGithubRepository,
+    { isSuccess: isSuccessCreatedGithubRepository },
+  ] = useCreateGithubRepositoryMutation()
+
+  useEffect(() => {
+    if (reposLoaded && repos.repositories) {
+      setRepositoryOwner(repos?.repositories[0].owner.login)
+      setRepositories(repos)
+    }
+  }, [repos, reposLoaded])
+
+  useEffect(() => {
+    if (
+      isSuccessCreatedGithubRepository &&
+      !localStorage?.githubIntegrationSetupFromFlagsmith
```

Comment:
Can we make this a variable rather than having this everywhere, it will also help if it was typed as it's not clear what this is

Reply:
Sure, Originally it was used once, the variable was created and typed.